components/cards/vendorCard.jsx

```diff
@@ -33,11 +32,11 @@ export default function VendorCard({
       </BookmarkButtonStyle>
       <h2>{cardTitle}</h2>
       <div className="cardContentWrapper">
-        <Image src={cardImage} alt="venue" width={100} height={100} />
+        <Image src={cardImage} alt="venue" height={100} width={100} />
         <div>
-          <a href="">Pricelist</a>
-          <a href="">Gallery</a>
-          <a href="">Contact</a>
+          <a href="/priceListVenueElbdeckEventLocation">Price list</a>
+          <a href="/carouselVenueElbdeckEventLocation">Gallery</a>
+          <a href="/contactVenueElbdeckEventLocation">Contact</a>
```

Comment:
This solution works for a single location, but it does not really scale.
Imagine having to create react components for hundreds of venues. You'd have to copy many lines of react component code that would be very hard to maintain.

I'd suggest the following.

Store your detail data somewhere where it is easy to edit and access e.g. `venues.json`

```json
[
  {
    "slug": "elbdeck-event-location",
    "name": "ELBDECK EVENT LOCATION",
    "images": [
       { "src": "/images/carouselVenue1Pic1.jpeg", "alt": "A very nice picture of the elbdeck!" },
       { "src": "/images/carouselVenue1Pic2.jpeg", "alt": "Another very nice picture of the elbdeck!" },
       …
    ],
    "phone": "tel:04012345",
    …
  },
  {
    "slug": "another-event-location",
    "name": "ANOTHER EVENT LOCATION",
    …
  },
  …
]
```

Then make your pages dynamic by using next.js dynamic routes:

file: `pages/venues/[slug]/price-list.js` -> example url: `/venues/elbdeck-event-location/price-list`
file: `pages/venues/[slug]/gallery.js` -> example url: `/venues/elbdeck-event-location/gallery`
file: `pages/venues/[slug]/contact.js` -> example url: `/venues/elbdeck-event-location/contact`


Folder or files with names in square brackets `[]` become dynamic. You can use the value provided by the url in your components multiple methods. Since your venue data is static you can use any of these (in any case you can work with venue data imported from the json file):

1. access venue data in [`getStaticPaths`](https://nextjs.org/docs/basic-features/data-fetching#getstaticpaths-static-generation) / [`getStaticProps`](https://nextjs.org/docs/basic-features/data-fetching#getstaticprops-static-generation): pages will be statically generated (best performance, ideal for this case, medium to high complexity)
2. access venue data in [`getServerSideProps`](https://nextjs.org/docs/basic-features/data-fetching#getserversideprops-server-side-rendering): data will be gathered on the server whenever a client requests a page (ok performance, medium to high complexity)
3. `const { query } = useRouter()` in your component and then `find()` the correct venue (ok performance, low complexity)

The great thing about building your app this way is, that to add a new venue all you'd have to is add more data to your json file. No more creating new components that have code duplication in them. ;)

Reply:
hi Jan, thank you for the suggestions. I have been searching for this since yesterday. On Friday, Martin suggested me some keywords to google  about this , but now I have the solutions already HE HE 👯